sdk/src/azure/iot/az_iot_adu_client.c

```diff
@@ -116,6 +117,8 @@ const uint8_t azure_iot_adu_root_key_n[385]
         0x77, 0x3e, 0x40, 0x87, 0x18, 0xc9, 0xab, 0xd9, 0xf7, 0x79 };
 const uint8_t azure_iot_adu_root_key_e[3] = { 0x01, 0x00, 0x01 };
 
+uint8_t azure_iot_step_id_scratch_buffer[7];
```

Comment:
This won't work if multi-threaded systems. If memory is required, we need to either:
- allocate from the stack (7 bytes is not a lot).
- require the app to pass it into the APIs that need it
- require the app to pass it into the client / handle initializer.